### ISSUE TITLE
Button submit

### ISSUE THREAD

**Issue by ezeparziale** [wtforms/wtforms#784]
Any reason for not having a button element with type="submit"?"

I created a new widget

```
class Button:
    html_params = staticmethod(html_params)
    validation_attrs = ["required"]

    def __init__(self, input_type=None):
        if input_type is not None:
            self.input_type = input_type

    def __call__(self, field, **kwargs):
        kwargs.setdefault("id", field.id)
        kwargs.setdefault("type", self.input_type)
        flags = getattr(field, "flags", {})
        for k in dir(flags):
            if k in self.validation_attrs and k not in kwargs:
                kwargs[k] = getattr(flags, k)
        return Markup("<button %s>%s</button>" % (self.html_params(name=field.name, **kwargs), field.label.text))

class SubmitButton(Button):
    input_type = "submit"

    def __call__(self, field, **kwargs):
        return super().__call__(field, **kwargs)
    

class SubmitButtonField(BooleanField):
    widget = SubmitButton()
```

```
class CreateForm(Form):
    submit = SubmitField("Edit")
    submit_button = SubmitButtonField("Edit")
```

**Comment by ThiefMaster** [comment]
`type="submit"` is already the (not very great) default for a `<button>` element...

**Comment by azmeuk** [comment]
This makes me think that `SubmitField` currently inherits from `BooleanField`:

https://github.com/pallets-eco/wtforms/blob/224a467c71477c10ddb180f09f803b436d959a10/src/wtforms/fields/simple.py#L127

```python
>>> from wtforms import Form, SubmitField
>>> from werkzeug.datastructures import MultiDict
>>> class MyForm(Form):
...     submit = SubmitField()
...     
>>> form = MyForm(MultiDict({"submit": "submit-intent"}))
>>> form.submit.data
True
```
This can be surprising (why is this not the value of the submit button?). I suppose this is historical, due to the fact that for `<input type="submit">`, the display label IS the value of the input, and that the label of a button is generally not a very interesting data, so we just want to know if the button has been clicked or not.

`<button type="submit">` does not have this constraint and can have a value different than its label `<button type="submit" value="submit-intent">Do this</button>`. This can be useful for instance to display several buttons with the same name, and different values, to represent different intents (save, save and add a new one, delete, reset etc.). To benefit from this, that would be interesting to associate a widget button to a field that would be more of a `StringField`.

Whether it is SubmitField changing its parent class, or a new field, I am not sure.

**Comment by ThiefMaster** [comment]
> Whether it is SubmitField changing its parent class

Please don't. I think such a change where it suddenly starts returning a string (which would also be user-controlled!) would be likely to break applications in code paths that are unlikely to be covered with tests that assume it's a boolean... I guess in most cases it would still be fine because it'd be None/True -> None->non-empty string, but I'm sure you have some "less pythonic" developers who did `form.submit.data == True`, `form.submit.data is True`, etc..

**Comment by azmeuk** [comment]
Fair point. The alternative would be adding a new field (say `ButtonField`) tied to a new button widget (say `ButtonWidget`). Would that sound better?
